zerver/worker/queue_processors.py

```diff
@@ -480,6 +482,17 @@ def get_bot_api_client(self, user_profile):
 
     def get_bot_handler(self, service):
         # type: (Service) -> Any
+
+        # Convert unicode service.name to string.
+        str_service_name = unicodedata.normalize('NFKD', service.name).encode('ascii', 'ignore')
+
+        # Assert if this service is present in EMBEDDED_BOT_SERVICES.
+        assert EmbeddedBotService(str_service_name) in EMBEDDED_BOT_SERVICES
+
+        for embedded_bot_service in EMBEDDED_BOT_SERVICES:
+            if embedded_bot_service.name == service.name:
+                assert embedded_bot_service.IS_ALLOWED is True
+
         bot_module_name = 'zulip_bots.bots.%s.%s' % (service.name, service.name)
         bot_module = importlib.import_module(bot_module_name)  # type: Any
         return bot_module.handler_class()
```

Comment:
Yeah, I planned on adding test-coverage for these new additions (that is why I didn't put this PR up for review yet). But definitely! it makes much more sense to shift it to `zerver/lib/bot_lib.py` and then test it.